### PR TITLE
Introduce the Weingarten map.

### PR DESCRIPTION
**Pull request by kellertuer**
We could check whether

- [x] Documentation
- [x] Tests
- [x] The News.md is a bit outdated (updating this was not so much fun).

### CODE REVIEW

File: src/ManifoldsBase.jl [hunk 1]
```diff
@@ -907,6 +907,39 @@ Converts a size given by `Tuple{N, M, ...}` into a tuple `(N, M, ...)`.
...
+of the embedded submanifold ``\mathcal M``, where we denote the embedding by ``\mathcal E``.
+
+One interpretation can also be given by looking at the differential of the [`project`](@ref)ion
+``\operatorname{proj}_{T_p\mathcal M}\colon \mathcal E \to T_p\mathcal M`` with respect to the base point ``p``,
+i.e. defining
+```math
+\mathcal P_X \coloneqq D_p\operatorname{proj}_{T_p\mathcal M}(Y)[X],
+\qquad Y \in \mathcal E, X \in T_p\mathcal M,
+```
+the Weingarten map can be written as ``\mathcal W_p(X,V) = \mathcal P_X(V)``.
```
Comment (mateuszbaran) [inline src/ManifoldsBase.jl:932]:
```suggestion
The Weingarten map can be defined by restricting the differential of the orthogonal [`project`](@ref)ion
``\operatorname{proj}_{T_p\mathcal M}\colon T_p \mathcal E \to T_p\mathcal M`` with respect to the base point ``p``,
i.e. defining
```math
\mathcal P_X \coloneqq D_p\operatorname{proj}_{T_p\mathcal M}(Y)[X],
\qquad Y \in T_p \mathcal E, X \in T_p\mathcal M,
```
the Weingarten map can be written as ``\mathcal W_p(X,V) = \mathcal P_X(V)``.
```
I think this is a bit more clear.

Comment (mateuszbaran) [reply src/ManifoldsBase.jl:932]: Hm, github suggestions don't work well with math formulas.

Comment (kellertuer) [reply src/ManifoldsBase.jl:932]: HM I think you maybe mixed up the code environments? Maybe do the outer - suggestion one – with more backticks - 4 or so?

File: src/ManifoldsBase.jl [hunk 2]
```diff
@@ -907,6 +907,39 @@ Converts a size given by `Tuple{N, M, ...}` into a tuple `(N, M, ...)`.
 """
 Base.@pure size_to_tuple(::Type{S}) where {S<:Tuple} = tuple(S.parameters...)
 
+@doc raw"""
+    Weingarten!(M, Y, p, X, V)
+
+Compute the Weingarten map ``\mathcal W_p\colon T_p\mathcal M × N_p\mathcal M \to T_p\mathcal M``
+in place of `Y`, see [`Weingarten`](@ref).
+"""
+Weingarten!(M::AbstractManifold, Y, p, X, A)
```
[the remark this answers is not in the document]
Comment (kellertuer) [reply src/ManifoldsBase.jl:916]: Yes I noticed along the way that A is not so nice. and V (maybe also W) is nicer here.

File: src/ManifoldsBase.jl (hunk 1 above)
Comment (mateuszbaran) [reply src/ManifoldsBase.jl:932]: Ah, yes, not enough backticks in the outer envoronment.

**Comment by kellertuer** [comment]
I am not sure why the report above says 3 lines are missing, which are from a Julia version pre 1.1, while the tests all pass. But besides that this should be good to go (even with updated News.md).

**Comment by kellertuer** [comment]
Of course I committed them! But you are right, I should probably also have pushed them 😴

File: NEWS.md [hunk 3]
```diff
@@ -5,7 +5,71 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
-## [0.14.0]
+## [0.14.10] xx/08/2023
+
```
Comment (mateuszbaran) [inline NEWS.md:9]:
```suggestion

### Added

```

**Review by mateuszbaran — APPROVED** [review]
OK, I think this looks fine.

**Pull request merged**